src/screens/RegisterOptionsScreen.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { StyleSheet, Text, View, Alert } from 'react-native';
+import { t } from 'ttag';
+import { useSelector } from 'react-redux';
+import { get } from 'lodash';
+import HathorHeader from '../components/HathorHeader';
+import NewHathorButton from '../components/NewHathorButton';
+import { COLORS } from '../styles/themes';
+import { NANO_CONTRACT_FEATURE_TOGGLE, REOWN_FEATURE_TOGGLE } from '../constants';
+
+const RegisterOptionsScreen = ({ navigation }) => {
+  const featureToggles = useSelector((state) => state.featureToggles);
+  const serverInfo = useSelector((state) => state.serverInfo);
+  const isReownEnabled = featureToggles[REOWN_FEATURE_TOGGLE] && get(serverInfo, 'nano_contracts_enabled', false);
+  const isNanoContractEnabled = featureToggles[NANO_CONTRACT_FEATURE_TOGGLE] && get(serverInfo, 'nano_contracts_enabled', false);
+
+  const getDescriptionText = () => {
+    if (!isNanoContractEnabled && !isReownEnabled) {
+      return t`You can register Tokens manually.`;
+    }
+
+    if (isNanoContractEnabled && !isReownEnabled) {
+      return t`You can register Tokens and Nano Contracts manually.`;
+    }
+
+    if (!isNanoContractEnabled && isReownEnabled) {
+      return t`You can register Tokens and use Reown manually.`;
+    }
+
+    return t`You can register Tokens, Nano Contracts or Reown manually.`;
+  };
+
+  const handleNanoContractNavigation = () => {
+    if (!isNanoContractEnabled) {
+      Alert.alert(
+        t`Feature Not Available`,
+        t`The nano contract feature is not enabled.`,
+        [{ text: t`OK` }]
+      );
+      return;
+    }
+    navigation.navigate('NanoContractRegisterScreen');
+  };
+
+  const handleReownNavigation = () => {
+    if (!isReownEnabled) {
+      Alert.alert(
+        t`Feature Not Available`,
+        t`The reown feature is not enabled.`,
+        [{ text: t`OK` }]
+      );
+      return;
+    }
+    navigation.navigate('ReownManual');
+  };
+
+  return (
+    <View style={styles.container}>
+      <HathorHeader
+        title={t`Register`}
+        onBackPress={() => navigation.goBack()}
+      />
+      <View style={styles.content}>
+        <Text style={styles.description}>
+          {getDescriptionText()}
+        </Text>
+        <View style={styles.buttonContainer}>
+          {isNanoContractEnabled && (
+            <NewHathorButton
+              title={t`Register Nano Contract`}
+              onPress={handleNanoContractNavigation}
+              style={styles.button}
+            />
+          )}
+          <NewHathorButton
+            title={t`Register Token`}
```

Comment:
question: Is this title still relevant or correct? If I understand correctly, this should be something along the lines of "Register with Manual Input" instead.

Reply:
It is still relevant, this is inside the screen after we press "Manual", so I don't think we should change the text, as it would be redundant